# All units are in SI
# ===================
# Time:     second, hz
# Angle:    radian
# Distance: meter

simulator:
  ros__parameters:
    save_rate: 1.0
    save_history: 4096
    headless: false

    ref_heading: 90.0  # For the GPS sensor to work

    objects:
      rover:
        type: urdf
        uri: package://mrover/urdf/rover/rover.urdf.xacro
        position: [ 0.0, 0.0, 0.1 ]
      world:
        type: urdf
        uri: package://mrover/urdf/world/world.urdf.xacro
      # tag_0:
      #   type: urdf
      #   uri: package://mrover/urdf/world/tag_0.urdf.xacro
      #   position: [ 11.0, 3.0, 0.7 ]
      # tag_1:
      #   type: urdf
      #   uri: package://mrover/urdf/world/tag_1.urdf.xacro
      #   position: [ 15.0, -14.0, 2.4 ]
      # bottle:
      #   type: urdf
      #   uri: package://mrover/urdf/world/bottle.urdf.xacro
      #   position: [ 16.0, 18.0, 0.5 ]
      rock1:
        type: urdf
        uri: package://mrover/urdf/world/small_rock.urdf.xacro
        position: [ 1.85, -1.85, 0.35 ]
      rock2:
        type: urdf
        uri: package://mrover/urdf/world/small_rock.urdf.xacro
        position: [ 2.9, 1.2, 0.35 ]
      rock3:
        type: urdf
        uri: package://mrover/urdf/world/small_rock.urdf.xacro
        position: [ -2.9, 2.9, 0.35 ]
      rock4:
        type: urdf
        uri: package://mrover/urdf/world/small_rock.urdf.xacro
        position: [ -4.3, -1.8, 0.35 ]
      rock5:
          type: urdf
          uri: package://mrover/urdf/world/small_rock.urdf.xacro
          position: [ -2.25, -5.25, 0.35 ]
      rock6:
          type: urdf
          uri: package://mrover/urdf/world/small_rock.urdf.xacro
          position: [ 2.5, 6.0, 0.35 ]


    # TODO(quintin): Avoid storing these in multiple config files
    ref_lat: 38.4225202
    ref_lon: -110.7844653
    ref_alt: 0.0
    world_frame: "map"
    rover_frame: "sim_base_link"
